### ISSUE TITLE
disable dataset augment  on hsv  parameters not work

### ISSUE THREAD

**Issue by Roger8** [ultralytics/ultralytics#9244]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Ultralytics YOLOv8.1.16 🚀 Python-3.9.10 torch-2.2.0+cpu CPU (Intel Core(TM) i5-10400F 2.90GHz)
engine\trainer: task=classify, mode=train, model=runs/classify/train9/weights/last.pt, data=./chars/, epochs=200, time=No
ne, patience=100, batch=64, imgsz=64, save=True, save_period=50, cache=False, device=None, workers=8, project=None, name=
train12, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rec
t=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=Fal
se, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou
=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=True, aug
ment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=Fal
se, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscr
ipt, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01
, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls
=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, **hsv_h=None, hsv_s=None, hsv_v=None**, degrees=0.0, transla
te=0.01, scale=0.08, shear=0, perspective=0, flipud=0.0, fliplr=0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=au
toaugment, erasing=0, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\classify\train1

![train_batch0](https://github.com/ultralytics/ultralytics/assets/7456643/c7a125e1-3bf6-4bca-9d12-ab4c87010568)

I have disabled hsv augment parameters, but it seems still working. 
Background color changing is not a need. I just want its **white background** . 
Did i set wrong train parameters?

**Comment by glenn-jocher** [comment]
@Roger8 hey there! 👋 It seems like you're trying to disable the HSV augmentation to maintain a white background in your images. To disable HSV augmentation in YOLOv8, setting `hsv_h`, `hsv_s`, and `hsv_v` to `None` directly in the command might not be the right approach.

Instead, ensure these parameters are set to `0` in your dataset's YAML configuration file, not during training command execution. Here's a quick example of how you can specify these parameters in your dataset config file (e.g., `chars.yaml`):

```yaml
# chars.yaml
train: ./data/train/
val: ./data/val/

# Augmentation settings
hsv_h: 0  # set hue augmentation to 0
hsv_s: 0  # set saturation augmentation to 0
hsv_v: 0  # set value augmentation to 0
...
```

After adjusting your dataset's YAML configuration file, proceed with your training command without specifying `hsv_h`, `hsv_s`, and `hsv_v` as they are now explicitly disabled through the config file.

If you've already done this and still encounter unexpected behavior, please double-check if your dataset configuration is correctly loaded during training. Feel free to reach out if the issue persists! 😊

**Comment by Roger8** [comment]
Hi @glenn-jocher ,thanks for your advice. I have tried , but it did not work . I don't know if the way which i used was wrong . 
Here is the code and cfg .
```python
from ultralytics import YOLO
# Load a model
model = YOLO('yolov8s-cls.pt') 
results = model.train(data="./chars/",cfg="ocr.yaml")
```
`ocr.yaml`
```yaml
batch: 64
imgsz: 64
# Augmentation settings
hsv_h: 0  # set hue augmentation to 0
hsv_s: 0  # set saturation augmentation to 0
hsv_v: 0  # set value augmentation to 0
fliplr: 0.0
```
The output 

```html
Ultralytics YOLOv8.1.16  Python-3.9.10 torch-2.2.0+cpu CPU (Intel Core(TM) i5-10400F 2.90GHz)
engine\trainer: task=classify, mode=train, model=None, data=./chars/, epochs=100, time=None, patienc
e=100, batch=64, imgsz=64, save=True, save_period=-1, cache=False, device=None, workers=8, project=N
one, name=train14, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, determinis
tic=True, single_cls=False,rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fract
ion=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0
, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=Fal
se, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=
False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=Fal
se, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=T
rue, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, si
mplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0
.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=1
2.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0, hsv_s=0, hsv_v=0, degrees=0.0, translate=0.1, s
cale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0,
 auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=ocr.yaml, tracker=botsort.yaml, save_
dir=runs\classify\train14
train: classtask\chars\train... found 7986 images in 16 classes ✅
val: classtask\chars\val... found 412 images in 16 classes ✅
test: None...
Overriding model.yaml nc=1000 with nc=16

                   from  n    params  module                                       arguments

  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]

  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]

  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]

  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]

  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, Tru
e]
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]

  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, Tru
e]
  7                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]

  8                  -1  1   1838080  ultralytics.nn.modules.block.C2f             [512, 512, 1, Tru
e]
  9                  -1  1    678416  ultralytics.nn.modules.head.Classify         [512, 16]

YOLOv8s-cls summary: 99 layers, 5101232 parameters, 5101232 gradients, 12.6 GFLOPs
Transferred 156/158 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir runs\classify\train14', view at http://localhost:6006/
train: Scanning classtask\chars\train... 7986 images, 0 corrupt: 100%|████
val: Scanning classtask\chars\val... 412 images, 0 corrupt: 100%|█████████
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'op
timizer', 'lr0' and 'momentum' automatically...
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 26 weight(decay=0.0), 27 weight(decay=0.
0005), 27 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
```
the train batch 0 image 
![train_batch0](https://github.com/ultralytics/ultralytics/assets/7456643/dd2caa7e-ecdb-46f5-8f32-05331cc55f6d)

Did i do some wrong ?

**Comment by glenn-jocher** [comment]
Hi @Roger8 👋, thanks for sharing the details of your attempt to disable HSV augmentations. Everything in your `ocr.yaml` configuration and training command looks correct. 

One thing to ensure is that the `ocr.yaml` file is correctly placed and accessible by your training script, and that your training script is indeed using this specific configuration file. The output log indicates that `hsv_h=0, hsv_s=0, hsv_v=0` as expected, which should essentially disable HSV augmentations.

If the background color changes are still happening, it might not be due to HSV augmentations but could instead be related to other image augmentation settings in your configuration or defaults, like `translate`, `scale`, `mosaic`, etc. Since you're aiming for white backgrounds and minimizing augmentations, you might also want to set `translate=0`, `scale=0`, `mosaic=0`, and so on, to see if that maintains the original backgrounds.

If the issue persists, it might help to double-check the raw input images to ensure they're as expected before augmentation, and review any other augmentation parameters that might be influencing the background. Sometimes, the devil is in the details!

Let us know how it goes! 😊

**Comment by Roger8** [comment]
hi @glenn-jocher , glad to hear your voice . The devil seems to be well hidden。 
First try : some other settings are added to `ocr.yaml`.
```yaml
batch: 64
imgsz: 64
# Augmentation settings
hsv_h: 0  # set hue augmentation to 0
hsv_s: 0  # set saturation augmentation to 0
hsv_v: 0  # set value augmentation to 0
degrees: 0.0 # (float) image rotation (+/- deg)
translate: 0.0 # (float) image translation (+/- fraction)
scale: 0.0 # (float) image scale (+/- gain)
shear: 0.0 # (float) image shear (+/- deg)
perspective: 0.0 # (float) image perspective (+/- fraction), range 0-0.001
flipud: 0.0 # (float) image flip up-down (probability)
fliplr: 0.0 # (float) image flip left-right (probability)
mosaic: 0.0 # (float) image mosaic (probability)
mixup: 0.0 # (float) image mixup (probability)
copy_paste: 0.0 # (float) segment copy-paste (probability)
auto_augment: randaugment # (str) auto augmentation policy for classification (randaugment, autoaugment, augmix)
erasing: 0.0 # (float) probability of random erasing during classification training (0-1)
crop_fraction: 1.0 # (float) image crop fraction for classification evaluation/inference (0-1)
```
output log 
```
Ultralytics YOLOv8.1.16 🚀 Python-3.9.10 torch-2.2.0+cpu CPU (Intel Core(TM) i5-10400F 2.90GHz)
[34m[1mengine\trainer: [0mtask=classify, mode=train, model=None, data=./chars/, epochs=100, time=None, 
patience=100, batch=64, imgsz=64, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, 
name=train20, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, 
single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False,
 freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, 
save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, 
stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, 
show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, 
show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, 
simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, 
warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, 
label_smoothing=0.0, nbs=64, hsv_h=0, hsv_s=0, hsv_v=0, degrees=0.0, translate=0.0, scale=0.0, shear=0.0, 
perspective=0.0, flipud=0.0, fliplr=0.0, mosaic=0.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, 
erasing=0.0, crop_fraction=1.0, cfg=ocr.yaml, tracker=botsort.yaml, save_dir=runs\classify\train20
```
Augmentations are not as expected. 

Second try:  i set this to be empty , it works . Until now ,the devil shows up. 
`auto_augment:  # .... `

**Comment by glenn-jocher** [comment]
Hey @Roger8! 👋 It looks like specifying `auto_augment` with `randaugment` was indeed the sneaky devil in your setup. By setting `auto_augment:`, you've effectively neutralized any augmentation policies that were still active, resulting in the desired behavior. Here's a simplified approach to ensure no augmentations are applied:

```yaml
# All other settings remain the same
auto_augment:  # explicitly left empty to disable auto augmentations
```

And your python snippet looks spot on! Just ensure your YAML file is being correctly referenced. Here's to keeping those backgrounds pristine white! 🎉 Let us know if anything else pops up.

**Comment by Roger8** [comment]
Thanks very much @glenn-jocher . By setting `auto_augment:,` , i have got what i wanted .  
Sincerely grateful for your help.

**Comment by glenn-jocher** [comment]
You're welcome, @Roger8! 🚀 I'm thrilled to hear that adjusting the `auto_augment` setting did the trick for you. If you have any more questions or run into further issues, feel free to reach out. Happy coding! 😊

**Comment by ScholliYT** [comment]
I had to specify these settings nested under the `augment` key in my `dataset.yaml`

```yaml
names:
  0: Car
  1: Tram
  2: Bike
  3: Person
path: /content/....
train: ./images/train/
val: ./images/val/

# Augmentation settings
augment:
  hsv_h: 0.0
  hsv_s: 0.0
  hsv_v: 0.0
```
